File: src/main/js/level1tests.js
var assert = require('./assert.js');
var code = require('./level1code.js');
var level1tests = {
    helloWorldTest: function () {
        assert.areEqual("Hello World!", code.helloWorld());
    },
    sumTest: function () {
        assert.areEqual(10, code.sum(0, 10));
        assert.areEqual(10, code.sum(2, 8));
        assert.areEqual(10, code.sum(6, 4));
        assert.areEqual(10, code.sum(5, 5));
    },
    productTest: function () {
        assert.areEqual(100, code.product(1, 100));
        assert.areEqual(100, code.product(2, 50));
        assert.areEqual(100, code.product(4, 25));
        assert.areEqual(100, code.product(5, 20));
    },
    squareTest: function () {
        assert.areEqual(1, code.square(1));
        assert.areEqual(4, code.square(2));
        assert.areEqual(16, code.square(4));
        assert.areEqual(25, code.square(5));
    },
    cubeTest: function () {
        assert.areEqual(1, code.cube(1));
        assert.areEqual(8, code.cube(2));
        assert.areEqual(27, code.cube(3));
        assert.areEqual(64, code.cube(4));
    },
    sqrtTest: function () {
        assert.areEqual(1, code.sqrt(1));
        assert.areEqual(2, code.sqrt(4));
        assert.areEqual(4, code.sqrt(16));
        assert.areEqual(5, code.sqrt(25));
    },
    quotientTest: function () {
        assert.areEqual(2, code.quotient(101, 50));
        assert.areEqual(4, code.quotient(102, 25));
        assert.areEqual(5, code.quotient(103, 20));
        assert.areEqual(10, code.quotient(104, 10));
    },
    remainderTest: function () {
        assert.areEqual(1, code.remainder(101, 50));
        assert.areEqual(2, code.remainder(102, 25));
        assert.areEqual(3, code.remainder(103, 20));
        assert.areEqual(4, code.remainder(104, 10));
    },
    assertEqualArrays: function (expected, actual) {
        for (var i = 0; i < expected.length; i++) {
            assert.areEqual(expected[i], actual[i]);
        }
        assert.areEqual(expected.length, actual.length);
    },
    squareAllTest: function () {
        var expected = [1, 4, 9, 16, 25];
        var actual = code.squareAll([1, 2, 3, 4, 5]);
        this.assertEqualArrays(expected, actual);
    },
    cubeAllTest: function () {
        var expected = [1, 8, 27, 64, 125];
        var actual = code.cubeAll([1, 2, 3, 4, 5]);
        this.assertEqualArrays(expected, actual);
    },
    productAllTest: function () {
        var expected = [2, 6, 12, 20, 30];
        var actual = code.productAll([1, 2, 3, 4, 5], [2, 3, 4, 5, 6]);
        this.assertEqualArrays(expected, actual);
    },
    sumAllTest: function () {
        var actual = code.sumAll([1, 9, 2, 8, 3, 7, 4, 6, 5, 5, 50]);
        assert.areEqual(100, actual);
    },
    reverseTest: function () {
        var expected = [1, 5, 1, 2, 1, 9, 1, 3];
        var actual = code.reverse([3, 1, 9, 1, 2, 1, 5, 1]);
        this.assertEqualArrays(expected, actual);
    }
};
module.exports = level1tests;

File: src/main/scala/Level1Tests.scala
object Level1Tests {

    @Rank(1)
    def helloWorldTest() {
        Assert.areEqual("Hello World!", Level1Code.helloWorld());
    }
    @Rank(2)
    def sumTest(): Unit = {
        Assert.areEqual(10, Level1Code.sum(0, 10))
        Assert.areEqual(10, Level1Code.sum(2, 8))
        Assert.areEqual(10, Level1Code.sum(6, 4))
        Assert.areEqual(10, Level1Code.sum(5, 5))
    }
    @Rank(3)
    def productTest(): Unit = {
        Assert.areEqual(100, Level1Code.product(1, 100))
        Assert.areEqual(100, Level1Code.product(2, 50))
        Assert.areEqual(100, Level1Code.product(4, 25))
        Assert.areEqual(100, Level1Code.product(5, 20))
    }
    @Rank(4)
    def squareTest(): Unit = {
        Assert.areEqual(1, Level1Code.square(1))
        Assert.areEqual(4, Level1Code.square(2))
        Assert.areEqual(16, Level1Code.square(4))
        Assert.areEqual(25, Level1Code.square(5))
    }
    @Rank(5)
    def cubeTest(): Unit = {
        Assert.areEqual(1, Level1Code.cube(1))
        Assert.areEqual(8, Level1Code.cube(2))
        Assert.areEqual(27, Level1Code.cube(3))
        Assert.areEqual(64, Level1Code.cube(4))
    }
    @Rank(6)
    def squareRootOfPerfectSquareTest(): Unit = {
        Assert.areEqual(Some(1), Level1Code.squareRootOfPerfectSquare(1))
        Assert.areEqual(Some(2), Level1Code.squareRootOfPerfectSquare(4))
        Assert.areEqual(Some(4), Level1Code.squareRootOfPerfectSquare(16))
        Assert.areEqual(Some(5), Level1Code.squareRootOfPerfectSquare(25))
        Assert.areEqual(None, Level1Code.squareRootOfPerfectSquare(2))
        Assert.areEqual(None, Level1Code.squareRootOfPerfectSquare(-25))
    }
    @Rank(7)
    def quotientTest(): Unit = {
        Assert.areEqual(2, Level1Code.quotient(101, 50))
        Assert.areEqual(4, Level1Code.quotient(102, 25))
        Assert.areEqual(5, Level1Code.quotient(103, 20))
        Assert.areEqual(10, Level1Code.quotient(104, 10))
    }
    @Rank(8)
    def remainderTest(): Unit = {
        Assert.areEqual(1, Level1Code.remainder(101, 50))
        Assert.areEqual(2, Level1Code.remainder(102, 25))
        Assert.areEqual(3, Level1Code.remainder(103, 20))
        Assert.areEqual(4, Level1Code.remainder(104, 10))
    }
    @Rank(9)
    def squareAllTest(): Unit = {
        Assert.areEqual(Array[Int](1, 4, 9, 16, 25).toSeq, Level1Code.squareAll(Array[Int](1, 2, 3, 4, 5)).toSeq)
    }
    @Rank(10)
    def cubeAllTest(): Unit = {
        Assert.areEqual(Array[Int](1, 8, 27, 64, 125).toSeq, Level1Code.cubeAll(Array[Int](1, 2, 3, 4, 5)).toSeq)
    }
    @Rank(11)
    def productAllTest(): Unit = {
        Assert.areEqual(Array[Int](2, 6, 12, 20, 30).toSeq, Level1Code.productAll(Array[Int](1, 2, 3, 4, 5), Array[Int](2, 3, 4, 5, 6)).toSeq)
    }
    @Rank(12)
    def sumAllTest(): Unit = {
        Assert.areEqual(100, Level1Code.sumAll(Array[Int](1, 9, 2, 8, 3, 7, 4, 6, 5, 5, 50)))
    }
    @Rank(13)
    def reverseTest(): Unit = {
        Assert.areEqual(Array[Int](1, 5, 1, 2, 1, 9, 1, 3).toSeq, Level1Code.reverse(Array[Int](3, 1, 9, 1, 2, 1, 5, 1)).toSeq)
    }

}
